packages/duckdb/index.cjs

```diff
@@ -185,17 +185,27 @@ module.exports.getRunner = async (opts, directory) => {
 	return async (queryContent, queryPath, batchSize) => {
 		// Filter out non-sql files
 		if (!queryPath.endsWith('.sql')) return null;
-		return runQuery(
-			queryContent,
-			{ ...opts, filename: path.join(directory, opts.filename) },
-			batchSize
-		);
+		// Determine filename based on string
+		const filename =
+			opts.filename.startsWith('md:?motherduck_token=') || opts.filename == ':memory:'
```

Comment:
I wonder if a more robust way is to assume that it is an absolute path by default, and only revert to relative if certain conditions are met.

https://duckdb.org/docs/connect

It seems there are three possibilities in general
- An existing file
- In memory - `:memory:`
- If no file is specified, one will be created (the duckdb CLI creates an in-memory file , for example)